operators/pkg/controller/elasticsearch/driver/downscale_invariants.go

```diff
@@ -0,0 +1,70 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package driver
+
+import (
+	"github.com/elastic/cloud-on-k8s/operators/pkg/apis/elasticsearch/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/label"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/reconcile"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/sset"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/utils/k8s"
+	appsv1 "k8s.io/api/apps/v1"
+)
+
+const (
+	OneMasterAtATimeInvariant        = "A master node is already in the process of being removed"
+	AtLeastOneRunningMasterInvariant = "Cannot remove the last running master node"
+)
+
+// DownscaleInvariants restricts downscales to perform in a single reconciliation attempt:
+// - remove a single master at once
+// - don't remove the last living master node
+type DownscaleInvariants struct {
+	// masterRemoved indicates whether a master node is in the process of being removed already.
```

Comment:
second naming nit, maybe `removalInProgress` or something similar since it is something currently happening rather than something in the past?